lib/realtime/api/presence.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Realtime.Api.Presence do
+  @moduledoc """
+  Defines the Presence schema
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+  @derive {Jason.Encoder, only: [:inserted_at, :updated_at, :id, :channel_id]}
+
+  @type t :: %__MODULE__{}
+
+  @schema_prefix "realtime"
+  schema "presences" do
+    field(:check, :boolean, default: false)
+    timestamps()
+
+    belongs_to(:channel, Realtime.Api.Channel)
+  end
+
+  def changeset(broadcast, attrs) do
+    broadcast
+    |> cast(attrs, [:check, :inserted_at, :updated_at, :channel_id])
+    |> put_timestamp(:updated_at)
+    |> maybe_put_timestamp(:inserted_at)
+  end
+
+  def check_changeset(broadcast, attrs) do
+    broadcast
+    |> change()
+    |> put_change(:check, attrs[:check])
```

Comment:
```suggestion
    change(broadcast, check: attrs[:check])
```

or given the internal vs external discussion above, I'd treat it as internal data:

```elixir
def check_changeset(broadcast, check) when is_boolean(check) do
  change(broadcast, check: check)
end
```

or remove this function and do everything at the call site.

Reply:
will tackle on a different PR so all changes are captured there